### PR TITLE
RM-7187: nunit3 migration - manual changes

### PR DESCRIPTION
**Pull request by dkudernatsch**

### CODE REVIEW

File: SecurityManager/Core.UnitTests/AclTools/Expansion/HtmlTagWriter/HtmlTagWriterTest.cs [hunk 1]
```diff
@@ -117,25 +117,29 @@ public void SpecificTagsTest ()
...
 
     [Test]
-    [ExpectedException (typeof (XmlException), ExpectedMessage = "Wrong closing tag in HTML: Expected abc but was xyz.")]
     public void NonMatchingEndTagTest ()
     {
       using (var htmlWriter = new SecurityManager.AclTools.Expansion.HtmlTagWriter.HtmlTagWriter (TextWriter.Null, false))
       {
         htmlWriter.Tag("abc");
-        htmlWriter.TagEnd ("xyz");
+        Assert.That(
```
Comment (Stefan-Ilic) [inline SecurityManager/Core.UnitTests/AclTools/Expansion/HtmlTagWriter/HtmlTagWriterTest.cs]: Format

File: Remotion/Data/DomainObjects.UnitTests/Persistence/Rdbms/SqlServer/IntegrationTests/SqlProviderSaveExistingTest.cs [hunk 2]
```diff
@@ -69,7 +69,7 @@ public void SaveAllSimpleDataTypes ()
       Assert.That (GetPropertyValue (savedClassWithAllDataTypes, typeof (ClassWithAllDataTypes), "Int32Property"), Is.EqualTo (2147483647));
       Assert.That (GetPropertyValue (savedClassWithAllDataTypes, typeof (ClassWithAllDataTypes), "Int64Property"), Is.EqualTo (9223372036854775807));
       Assert.That (GetPropertyValue (savedClassWithAllDataTypes, typeof (ClassWithAllDataTypes), "SingleProperty"), Is.EqualTo (6789.321f));
-      Assert.That (GetPropertyValue (savedClassWithAllDataTypes, typeof (ClassWithAllDataTypes), "StringProperty"), Is.EqualTo ("abcdef���"));
+      Assert.That (GetPropertyValue (savedClassWithAllDataTypes, typeof (ClassWithAllDataTypes), "StringProperty"), Is.EqualTo ("abcdeföäü"));
```
[the remark this answers is not in the document]
Comment (Stefan-Ilic) [reply Remotion/Data/DomainObjects.UnitTests/Persistence/Rdbms/SqlServer/IntegrationTests/SqlProviderSaveExistingTest.cs]: Ah, now I see the encodiing, the previous file view in Github didn't show anything

File: Remotion/Core/Reflection.UnitTests/TypeAdapterConverterTest.cs [hunk 3]
```diff
@@ -1,4 +1,4 @@
-﻿// This file is part of the re-motion Core Framework (www.re-motion.org)
+// This file is part of the re-motion Core Framework (www.re-motion.org)
```
[the remark this answers is not in the document]
Comment (MichaelKetting) [reply Remotion/Core/Reflection.UnitTests/TypeAdapterConverterTest.cs]: BOM

File: Remotion/Data/DomainObjects.Web.Test/Data.DomainObjects.Web.Test.csproj [hunk 4]
```diff
@@ -55,8 +55,7 @@
     <DebugSymbols>true</DebugSymbols>
     <FileAlignment>4096</FileAlignment>
     <NoStdLib>false</NoStdLib>
-    <NoWarn>
-    </NoWarn>
+
```
[the remark this answers is not in the document]
Comment (MichaelKetting) [reply Remotion/Data/DomainObjects.Web.Test/Data.DomainObjects.Web.Test.csproj]: Good catch. Sematnically, that's not a change. Still extra noise but the question is if fixing this is worth it, considering we're going for SDK-style files in a couple of months

File: Remotion/Security/UnitTests/SecurityClientTests/CheckMethodAccessTest_WithMethodInfo.cs [hunk 5]
```diff
@@ -109,50 +107,55 @@ public void Test_WithoutRequiredPermissionsAndWithinSecurityFreeSection_ShouldTh
...
-        _securityClient.CheckMethodAccess (_testHelper.SecurableObject, _methodInfo);
+        Assert.That (
+            () => _securityClient.CheckMethodAccess (_testHelper.SecurableObject, _methodInfo),
+            Throws.ArgumentException
+                .With.Message.EqualTo ("The member 'InstanceMethod' does not define required permissions.\r\nParameter name: requiredAccessTypeEnums"));  
       }
-
-      _testHelper.VerifyAll ();
+      
+      _testHelper.VerifyAll();
```
[the remark this answers is not in the document]
Comment (MichaelKetting) [reply Remotion/Security/UnitTests/SecurityClientTests/CheckMethodAccessTest_WithMethodInfo.cs]: Interesting. Good catches. I most definitely prefer not to have any unrelated formating changes in the files.

File: Remotion/Web/UnitTests/Core/ExecutionEngine/WxeFunctionTests/WxeFunctionTest.cs [hunk 6]
```diff
@@ -64,7 +64,7 @@ public void GetFunctionToken_MissingFunctionToken ()
       TestFunction rootFunction = new TestFunction();
 
       Assert.That (
-          () => Dev.Null = rootFunction.FunctionToken,
+          () => rootFunction.FunctionToken,
```
[the remark this answers is not in the document]
Comment (MichaelKetting) [reply Remotion/Web/UnitTests/Core/ExecutionEngine/WxeFunctionTests/WxeFunctionTest.cs]: no, Dev.Null is just syntax needed for doing property-reads. In the lambda-syntax those are no longer needed.

File: SecurityManager/Core.UnitTests/AclTools/Expansion/HtmlTagWriter/HtmlTagWriterTest.cs (hunk 1 above)
Comment (MichaelKetting) [reply SecurityManager/Core.UnitTests/AclTools/Expansion/HtmlTagWriter/HtmlTagWriterTest.cs]: What's wrong here?

File: Remotion/Security/UnitTests/SecurityClientTests/CheckMethodAccessTest_WithMethodInfo.cs [hunk 7]
```diff
@@ -161,10 +164,13 @@ public void Test_WithPermissionProviderReturnedNullAndWithinSecurityFreeSection_
...
       {
-        _securityClient.CheckMethodAccess (_testHelper.SecurableObject, _methodInfo);
+        Assert.That (
+            () => _securityClient.CheckMethodAccess (_testHelper.SecurableObject, _methodInfo),
+            Throws.InvalidOperationException
+                .With.Message.EqualTo ("IPermissionProvider.GetRequiredMethodPermissions evaluated and returned null."));
       }
-
-      _testHelper.VerifyAll ();
+
```
[the remark this answers is not in the document]
Comment (MichaelKetting) [reply Remotion/Security/UnitTests/SecurityClientTests/CheckMethodAccessTest_WithMethodInfo.cs]: I think this step was manually done (need to check the history, manual und auto-steps should be separate commits)

The extra blank was an editor artifact. The extra line is kinda correc tin this case. The official style is to do a blank line after a block.

File: Remotion/Security/UnitTests/SecurityClientTests/CheckMethodAccessTest_WithMethodInfo.cs (hunk 5 above)
[the remark this answers is not in the document]
Comment (Stefan-Ilic) [reply Remotion/Security/UnitTests/SecurityClientTests/CheckMethodAccessTest_WithMethodInfo.cs]: I did see it enough times to judge that I didn't want to comment on it every time (gut feeling is 10-15 files) My suggestion would be to leave it as is now and gradually fix these in separate commits as we come across them

File: SecurityManager/Core.UnitTests/AclTools/Expansion/HtmlTagWriter/HtmlTagWriterTest.cs (hunk 1 above)
Comment (Stefan-Ilic) [reply SecurityManager/Core.UnitTests/AclTools/Expansion/HtmlTagWriter/HtmlTagWriterTest.cs]: In 99% of the files changed there's a space between `.That` and `(`.

Comment (MichaelKetting) [reply SecurityManager/Core.UnitTests/AclTools/Expansion/HtmlTagWriter/HtmlTagWriterTest.cs]: *blink* Do'oh!

File: Remotion/Mixins/Samples/Tutorial/T02_ParamList/UnitTests/CtorCallTest.cs [hunk 8]
```diff
@@ -25,28 +25,28 @@ namespace Remotion.Mixins.Samples.Tutorial.T02_ParamList.UnitTests
   [TestFixture]
   public class CtorCallTest
   {
-    [Test]
-    [ExpectedException (typeof (MissingMethodException))] // looks for a ctor with five string arguments
+    [Test] // looks for a ctor with five string arguments
```
[the remark this answers is not in the document]
Comment (dkudernatsch) [reply Remotion/Mixins/Samples/Tutorial/T02_ParamList/UnitTests/CtorCallTest.cs]: should the comment be on the Throws... line?

[the remark this answers is not in the document]
Comment (Stefan-Ilic) [reply Remotion/Mixins/Samples/Tutorial/T02_ParamList/UnitTests/CtorCallTest.cs]: I'd put it in the line next to the method call

[the remark this answers is not in the document]
Comment (dkudernatsch) [reply Remotion/Mixins/Samples/Tutorial/T02_ParamList/UnitTests/CtorCallTest.cs]: `Assert.That (  // looks for a ctor with five string arguments`
so?

[the remark this answers is not in the document]
Comment (MichaelKetting) [reply Remotion/Mixins/Samples/Tutorial/T02_ParamList/UnitTests/CtorCallTest.cs]: next to or in the line above the lambda or the line above the Assert.That() statement itself. Don't think we have too many comments, correct?

[the remark this answers is not in the document]
Comment (MichaelKetting) [reply Remotion/Mixins/Samples/Tutorial/T02_ParamList/UnitTests/CtorCallTest.cs]:
```
// looks
Assert.That (
    () => ...,
    Throws...);
```
```
Assert.That (
    // looks
    () => ...,
    Throws...);
```

```
Assert.That (
    () => ...,  // looks
    Throws...);
```

[the remark this answers is not in the document]
Comment (Stefan-Ilic) [reply Remotion/Mixins/Samples/Tutorial/T02_ParamList/UnitTests/CtorCallTest.cs]: I saw only 2 comments in this file and didn't see them anywhere else

File: Remotion/Mixins/UnitTests/Core/CodeGeneration/IntegrationTests/MixedTypeCodeGeneration/InstantiationTest.cs [hunk 9]
```diff
@@ -18,6 +18,8 @@
 using System.Reflection;
 using System.Reflection.Emit;
 using NUnit.Framework;
+using NUnit.Framework.Constraints;
+using Remotion.FunctionalProgramming;
```
[the remark this answers is not in the document]
Comment (dkudernatsch) [reply Remotion/Mixins/UnitTests/Core/CodeGeneration/IntegrationTests/MixedTypeCodeGeneration/InstantiationTest.cs]: probably just auto inserted while manually editing files, i will remove those

File: Remotion/Mixins/UnitTests/Mixins.UnitTests.csproj [hunk 10]
```diff
@@ -12,6 +12,7 @@
     <AssemblyName>Remotion.Mixins.UnitTests</AssemblyName>
     <TargetFrameworkVersion>v4.6.1</TargetFrameworkVersion>
     <FileAlignment>512</FileAlignment>
+    <NoWarn>618</NoWarn>
```
[the remark this answers is not in the document]
Comment (dkudernatsch) [reply Remotion/Mixins/UnitTests/Mixins.UnitTests.csproj]: those should have been removed in d9d5a5086bb54fb581d825101538ed6deee713c2, not  sure how they dodged a global search and replace

[the remark this answers is not in the document]
Comment (Stefan-Ilic) [reply Remotion/Mixins/UnitTests/Mixins.UnitTests.csproj]: Alright, keep in mind that I didn't comment on every occurrence I found

File: Remotion/Security/UnitTests/SecurityClientTests/CheckMethodAccessTest_WithMethodInfo.cs (hunk 5 above)
[the remark this answers is not in the document]
Comment (dkudernatsch) [reply Remotion/Security/UnitTests/SecurityClientTests/CheckMethodAccessTest_WithMethodInfo.cs]: I went over all manually changed files again and i think i have now fixed all the formatting errors

File: Remotion/Data/DomainObjects.PerformanceTests/InstantiationTest.cs [hunk 11]
```diff
@@ -30,7 +30,7 @@ public class InstantiationTest : DatabaseTest
     [Test]
     public void GetObjectReference ()
     {
-      Console.WriteLine ("Expected average duration of GetObjectReference on reference system: ~15 �s (release build), ~20 �s (debug build)");
+      Console.WriteLine ("Expected average duration of GetObjectReference on reference system: ~15 �s (release build), ~20 �s (debug build)");
```
Comment (Stefan-Ilic) [inline Remotion/Data/DomainObjects.PerformanceTests/InstantiationTest.cs]: Please fix the encodings as we talked about

**Comment by Stefan-Ilic** [comment]
NUnit3 has been added to `master` with d8503c097b62248dabc92a4c1d2bb0e0ffdaffd7

**Pull request closed without merging**